zio-http/src/main/scala/zhttp/http/middleware/Web.scala

```diff
@@ -171,6 +172,24 @@ private[zhttp] trait Web extends Cors with Csrf with Auth with HeaderModifier[Ht
       _ => middleware,
       _ => Middleware.identity,
     )
+
+  /**
+   * Removes the trailing slash from the path.
+   */
+  final def dropTrailingSlash: HttpMiddleware[Any, Nothing] =
+    Middleware.identity[Request, Response].contramap[Request](_.dropTrailingSlash).when(_.url.queryParams.isEmpty)
+
+  /**
+   * Permanent redirect if the trailing slash is present in the request URL.
+   */
+  final def redirectTrailingSlash(permanent: Boolean): HttpMiddleware[Any, Nothing] =
+    Middleware.ifThenElse[Request](_.url.path.trailingSlash)(
+      req =>
+        dropTrailingSlash ++ Middleware.fromHttp(
+          Http.response(Response.redirect(encode(req.dropTrailingSlash.url), isPermanent = permanent)),
+        ),
```

Comment:
```suggestion
        dropTrailingSlash ++ Middleware.redirect(req.dropTrailingSlash.url, permanent),
```
We can add `Middleware.redirect` as another Middleware. I think it's useful.

Reply:
added